### PR TITLE
ARROW-2236: [JS] Add more complete set of predicates

### PR DESCRIPTION
**Pull request by TheNeuralBit**
Adds not, gt, lt, and neq

### CODE REVIEW

File: js/test/unit/vector-tests.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import { TextEncoder } from 'text-encoding-utf-8';
 import Arrow from '../Arrow';
 import { type, TypedArray, TypedArrayConstructor, Vector } from '../../src/Arrow';
-import { packBools } from '../../src/util/bit'
```
Comment (trxcllnt) [inline js/test/unit/vector-tests.ts]: @TheNeuralBit ah, do we not export this util in the main package? if so, we can. that way we'll test the compiled code. On line 20 I think we should also do `const { type, Vector } = Arrow;`, and only import the interfaces from the source dir.

Comment (TheNeuralBit) [reply js/test/unit/vector-tests.ts]: Yeah good call. My syntax checker, [tsuquyomi](https://github.com/Quramy/tsuquyomi), complains about the `const { type, Vector } = Arrow;` approach so I shied away from it, but the tests run just fine.

**Comment by wesm** [comment]
When this is good to go / merged I can cut the new RC, let me know

**Review by trxcllnt — APPROVED** [review]
LGTM 👍

File: js/src/Arrow.externs.js [hunk 2]
```diff
@@ -79,6 +79,12 @@ Value.prototype.gteq;
 Value.prototype.lteq;
```
Comment (wesm) [inline js/src/Arrow.externs.js]: Is there any prior art for naming these methods in JavaScript? (in Python we have lt, le, gt, ge, eq, ne)

Comment (TheNeuralBit) [reply js/src/Arrow.externs.js]: Not that I'm aware of. We (CCRi) chose them arbitrarily when first implementing this. I wouldn't be opposed to renaming to those two-letter names to be more consistent and concise.

Comment (wesm) [reply js/src/Arrow.externs.js]: @trxcllnt do you have an opinion?

Comment (trxcllnt) [reply js/src/Arrow.externs.js]: @wesm I'd be partial to the two-letter names if it means we're more aligned with other data-science tools. But I'm also not the primary consumer of these APIs yet (we're focused on GPU dataframe ops), so I feel my vote should count less than @TheNeuralBit's here.

Comment (TheNeuralBit) [reply js/src/Arrow.externs.js]: ok I'll go ahead and update the names here. Thanks for the suggestion Wes, it's good to make these changes now.

**Review by wesm — APPROVED** [review]
+1

**Comment by wesm** [comment]
I'll get going on a 0.3.1 RC

**Comment by TheNeuralBit** [comment]
Thanks Wes! Anything I can do to help out with the release process?

**Comment by wesm** [comment]
Nope, it should be pretty straightforward. It might be nice to have the release script generate an e-mail template like https://github.com/apache/parquet-cpp/blob/master/dev/release/release-candidate#L257

**Pull request closed without merging**